folding-schemes/src/frontend/noname/utils.rs

```diff
@@ -0,0 +1,157 @@
+}
+
+impl<F: PrimeField> TryFrom<(String, Option<String>)> for NoNameIVCInputs<F> {
+    type Error = Error;
+
+    fn try_from(value: (String, Option<String>)) -> Result<Self, Error> {
+        let (ivc_public_inputs, external_inputs) = value;
+        let json_ivc_inputs = serde_json::from_str::<NoNameJSONIVCInput>(&ivc_public_inputs)
+            .map_err(|e| Error::JSONSerdeError(e.to_string()))?;
+
+        let json_external_inputs = match external_inputs {
+            Some(inputs) => Some(
+                serde_json::from_str::<NoNameJSONExternalInputs>(&inputs)
+                    .map_err(|e| Error::JSONSerdeError(e.to_string()))?,
+            ),
+            None => None,
+        };
+
+        let mut ivc_input = Vec::with_capacity(json_ivc_inputs.ivc_input.len());
+        for str_input in json_ivc_inputs.ivc_input.iter() {
+            let parsed_biguint = BigUint::from_str(str_input)
+                .map_err(|e| Error::BigIntConversionError(e.to_string()))?;
+            let field_element = F::from_le_bytes_mod_order(&parsed_biguint.to_bytes_be());
+            ivc_input.push(field_element);
+        }
+
+        let external_inputs = match json_external_inputs {
+            Some(inputs) => {
+                let mut external_inputs = Vec::with_capacity(inputs.external_inputs.len());
+                for str_input in inputs.external_inputs.iter() {
+                    let parsed_biguint = BigUint::from_str(str_input)
+                        .map_err(|e| Error::BigIntConversionError(e.to_string()))?;
+                    let field_element = F::from_le_bytes_mod_order(&parsed_biguint.to_bytes_be());
+                    external_inputs.push(field_element);
+                }
+                Some(external_inputs)
+            }
+            None => None,
+        };
+
+        Ok(NoNameIVCInputs {
+            ivc_input,
+            external_inputs,
+        })
+    }
+}
+
+impl<F: PrimeField> TryInto<(JsonInputs, JsonInputs)> for NoNameIVCInputs<F> {
+    type Error = Error;
+
+    fn try_into(self) -> Result<(JsonInputs, JsonInputs), Error> {
+        let ivc_inputs = serde_json::to_string(&NoNameJSONIVCInput::from(self.ivc_input))
+            .map_err(|e| Error::JSONSerdeError(e.to_string()))?;
+        let external_inputs = match self.external_inputs {
+            Some(inputs) => parse_inputs(
+                &serde_json::to_string(&NoNameJSONExternalInputs::from(inputs))
+                    .map_err(|e| Error::JSONSerdeError(e.to_string()))?,
+            )
+            .map_err(|e| Error::Other(format!("{:?}", e)))?,
+            None => parse_inputs(r#"{}"#).map_err(|e| Error::Other(format!("{:?}", e)))?,
+        };
+        Ok((
+            parse_inputs(&ivc_inputs).map_err(|e| Error::Other(format!("{:?}", e)))?,
+            external_inputs,
+        ))
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct NoNameExternalInputs {
+    pub external_inputs: Vec<usize>,
+}
```

Comment:
I'm not finding any usage of this struct in the repo, is that correct? Maybe it can be removed (?)

Reply:
Correct, removed it